server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,26 +342,137 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
+                          "Starting lookup [%s]:[%s]",
+                          lookupBean.getName(),
+                          container
+                      );
+                      if (container.getLookupExtractorFactory().start()) {
+                        LOG.info(
+                            "Started lookup [%s]:[%s]",
+                            lookupBean.getName(),
+                            container
+                        );
+                        return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
+                      } else {
+                        LOG.error(
+                            "Failed to start lookup [%s]:[%s]",
+                            lookupBean.getName(),
+                            container
+                        );
+                        return null;
+                      }
+                    }
+                )
+            );
+          }
+          final ListenableFuture<List<Map.Entry>> futureList = Futures.allAsList(futures);
+          try {
+            futureList.get()
+                      .stream()
+                      .filter(Objects::nonNull)
+                      .forEach(builder::put);
+
+            stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
+
+            executorService.shutdownNow();
+            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+              LOG.warn("Lookup loading didn't complete in 60 seconds!");
+            }
+          }
+          catch (InterruptedException e) {
+            LOG.error("Lookup loading interrupted. ");
+            Thread.currentThread().interrupt();
 
-      final List<LookupBean> lookupBeanList = lookupSnapshotTaker.pullExistingSnapshot();
-      for (LookupBean lookupBean : lookupBeanList) {
-        LookupExtractorFactoryContainer container = lookupBean.getContainer();
+          }
+          catch (Exception ex) {
+            futureList.cancel(true);
+            throw ex;
+          }
+        }
+        catch (Exception e) {
+          LOG.error(e, "Failed to finish lookup load process.");
+        }
+      } else {
+        LOG.info("No lookups to be loaded at this point");
+        stateRef.set(new LookupUpdateState(ImmutableMap.of(), ImmutableList.of(), ImmutableList.of()));
+      }
+  }
 
-        if (container.getLookupExtractorFactory().start()) {
-          builder.put(lookupBean.getName(), container);
+  private List<LookupBean> getLookupListFromCoordinator(String tier)
+  {
+    try {
+      final FullResponseHolder response = fetchLookupsForTier(tier);
+      List<LookupBean> lookupBeanList = new ArrayList<>();
+      if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+        LOG.error(
+            "Error while fetching lookup code from Coordinator with status[%s] and content[%s]",
+            response.getStatus(),
+            response.getContent()
+        );
+        return null;
+      } else {
```

Comment:
do you need an else block here since you `return` above? Removing the extra indentation makes it easier to read